sumologic/resource_sumologic_monitors_library_monitor.go

```diff
@@ -610,6 +647,19 @@ var timeRangeSchema = schema.Schema{
 	DiffSuppressFunc: SuppressEquivalentTimeDiff(false),
 }
 
+func getSloBurnRateTimeRangeSchema(triggerType string) *schema.Schema {
+	requiredWith := fmt.Sprintf("trigger_conditions.0.slo_burn_rate_condition.0.%s.0.burn_rate_threshold", triggerType)
+	conflictsWith := fmt.Sprintf("trigger_conditions.0.slo_burn_rate_condition.0.%s.0.burn_rate", triggerType)
+	return &schema.Schema{
+		Type:             schema.TypeString,
+		Optional:         true,
+		ValidateFunc:     validation.StringMatch(regexp.MustCompile(`^-?(\d)+[smhd]$`), "Time range must be in the format '-?\\d+[smhd]'. Examples: -15m, 1d, etc."),
```

Comment:
we should deduplicate this validation function, it's also used in timeRangeSchema

Reply:
updated, [commit](https://github.com/SumoLogic/terraform-provider-sumologic/pull/499/commits/88c52da73c41464be2c8873369b0b2ba207afa0e#diff-b749eb8ae5017ef0ed2545e094d7e38321f015a91b335265140abcf75fb61365R654:~:text=var-,timeRangeValidation,-%3D%20validation.)